src/engine/routing_algorithms/shortest_path.cpp

```diff
@@ -1,6 +1,11 @@
 #include "engine/routing_algorithms/shortest_path.hpp"
+
+#ifdef OSRM_EXTERNAL_MEMORY
+#include "routing/routing_base_offline.hpp"
```

Comment:
This is not really good to mix different code bases.
@TheMarex  I understand the point about placing code into cpp file, but we still need to move templated version into an impl header, remove references to offline code here and keep template instantiations in a cpp file to avoid rebuild.
So offline code could instantiate own implementations using private impl headers. Otherwise we again make wrong dependencies into OSRM code and block changes in separate projects.